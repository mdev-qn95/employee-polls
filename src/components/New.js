import { useState } from "react";
import { Container, Row, Col, Form, Button } from "react-bootstrap";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { addPoll } from "../store/actions/questionsAction";

const New = ({ dispatch }) => {
  const [firstVal, setFirstVal] = useState("");
  const [secondVal, setSecondVal] = useState("");
  const navigate = useNavigate();

  const firstOptionChangeHandle = (e) => {
    const value = e.target.value;
    setFirstVal(value);
  };

  const secondOptionChangeHandle = (e) => {
    const value = e.target.value;
    setSecondVal(value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addPoll(firstVal, secondVal));
    navigate("/");
  };

  return (
    <Container className="mt-5 new">
      <Row>
        <Col>
          <h2>Would You Rather</h2>
          <span className="gray">Create Your Own Poll</span>
          <Form onSubmit={handleSubmit} className="mt-5">
            <Form.Group controlId="user">
              <Form.Label className="bold">First Option</Form.Label>
              <Form.Control
                type="text"
                name="one"
                placeholder="Option One"
                value={firstVal}
                onChange={firstOptionChangeHandle}
              />
            </Form.Group>

            <Form.Group controlId="password" className="mt-5">
              <Form.Label className="bold">Second Option</Form.Label>
              <Form.Control
                type="text"
                name="two"
                placeholder="Option Two"
                value={secondVal}
                onChange={secondOptionChangeHandle}
              />
            </Form.Group>

            <Button variant="primary" type="submit" className="mt-5">
              Submit
            </Button>
          </Form>
        </Col>
      </Row>
    </Container>
  );
};

export default connect()(New);
