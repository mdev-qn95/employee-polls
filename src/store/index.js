import { configureStore } from "@reduxjs/toolkit";
import users from "./reducers/usersReducers";
import authed from "./reducers/authedReducers";
import questions from "./reducers/questionsReducers";

export const store = configureStore({
  reducer: {
    authed,
    users,
    questions,
  },
});
