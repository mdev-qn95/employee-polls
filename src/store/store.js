import { configureStore } from "@reduxjs/toolkit";
import authed from "./reducers/authedReducers";
import users from "./reducers/usersReducers";
import questions from "./reducers/questionsReducers";

export const store = configureStore({
  reducer: {
    authed,
    users,
    questions,
  },
});
