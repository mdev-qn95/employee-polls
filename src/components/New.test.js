import React from "react";
import { Provider } from "react-redux";
import { fireEvent, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import New from "./New";
import { store } from "../store";

describe("New", () => {
  it("will render the component", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <New />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("will display the value of the options", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <New />
        </Router>
      </Provider>
    );

    const optionOneEle = component.getByTestId("option-one");
    const optionTwoEle = component.getByTestId("option-two");

    fireEvent.change(optionOneEle, { target: { value: "Javascript" } });
    fireEvent.change(optionTwoEle, { target: { value: "Typescript" } });
    expect(optionOneEle.value).toBe("Javascript");
    expect(optionTwoEle.value).toBe("Typescript");
  });
});
