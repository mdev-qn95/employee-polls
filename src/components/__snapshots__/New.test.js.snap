// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New will render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 new container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h2>
              Would You Rather
            </h2>
            <span
              class="gray"
            >
              Create Your Own Poll
            </span>
            <form
              class="mt-5"
            >
              <div>
                <label
                  class="bold form-label"
                  for="user"
                >
                  First Option
                </label>
                <input
                  class="form-control"
                  data-testid="option-one"
                  id="user"
                  name="one"
                  placeholder="Option One"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="mt-5"
              >
                <label
                  class="bold form-label"
                  for="password"
                >
                  Second Option
                </label>
                <input
                  class="form-control"
                  data-testid="option-two"
                  id="password"
                  name="two"
                  placeholder="Option Two"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="mt-5 btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 new container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h2>
            Would You Rather
          </h2>
          <span
            class="gray"
          >
            Create Your Own Poll
          </span>
          <form
            class="mt-5"
          >
            <div>
              <label
                class="bold form-label"
                for="user"
              >
                First Option
              </label>
              <input
                class="form-control"
                data-testid="option-one"
                id="user"
                name="one"
                placeholder="Option One"
                type="text"
                value=""
              />
            </div>
            <div
              class="mt-5"
            >
              <label
                class="bold form-label"
                for="password"
              >
                Second Option
              </label>
              <input
                class="form-control"
                data-testid="option-two"
                id="password"
                name="two"
                placeholder="Option Two"
                type="text"
                value=""
              />
            </div>
            <button
              class="mt-5 btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
